islpy/__init__.py

```diff
@@ -687,6 +687,56 @@ def pw_get_aggregate_domain(self):
 
         return result
 
+    def pw_le(self, other):
+        """
+        :return: self <= other
+        """
+        if isinstance(other, int):
+            return self.le_set(self * 0 + other)
+        else:
+            assert isinstance(other, PwAff)
+            return self.le_set(other)
+
+    def pw_lt(self, other):
+        """
+        :return: self < other
+        """
+        if isinstance(other, int):
+            return self.lt_set(self * 0 + other)
+        else:
+            assert isinstance(other, PwAff)
+            return self.lt_set(other)
+
+    def pw_ge(self, other):
+        """
+        :return: self >= other
+        """
+        if isinstance(other, int):
+            return self.ge_set(self * 0 + other)
+        else:
+            assert isinstance(other, PwAff)
+            return self.ge_set(other)
+
+    def pw_gt(self, other):
+        """
+        :return: self >= other
+        """
+        if isinstance(other, int):
+            return self.gt_set(self * 0 + other)
+        else:
+            assert isinstance(other, PwAff)
+            return self.gt_set(other)
+
+    def pw_bool(self):
+        return NotImplementedError(
+                "Converting a PwAff to a boolean is nor supported.")
+
+    PwAff.__le__ = pw_le
+    PwAff.__lt__ = pw_lt
+    PwAff.__ge__ = pw_ge
+    PwAff.__gt__ = pw_gt
```

Comment:
One issue with this is that it can't extend to `==` and `!=`. I wonder whether that invalidates the idea...

Reply:
True. This is a problem for eg: SymPy where `==` and `!=` are structural equivalence and `Eq` gives an expression. Numpy however is consistent and implements `==` and `!=` and returns a numpy array of bools.